service/dap/server.go

```diff
@@ -2667,10 +2668,170 @@ func (s *Server) onReadMemoryRequest(request *dap.ReadMemoryRequest) {
 	s.sendNotYetImplementedErrorResponse(request.Request)
 }
 
-// onDisassembleRequest sends a not-yet-implemented error response.
-// Capability 'supportsDisassembleRequest' is not set 'initialize' response.
+// onDisassembleRequest handles 'disassemble' requests.
+// Capability 'supportsDisassembleRequest' is set in 'initialize' response.
 func (s *Server) onDisassembleRequest(request *dap.DisassembleRequest) {
-	s.sendNotYetImplementedErrorResponse(request.Request)
+	// TODO(suzmue): microsoft/vscode#129655 is discussing the difference between
+	// memory reference and instructionPointerReference, which are currently
+	// being used interchangeably by vscode.
+	addr, err := strconv.ParseInt(request.Arguments.MemoryReference, 0, 64)
+	if err != nil {
+		s.sendErrorResponse(request.Request, UnableToDisassemble, "Unable to disassemble", err.Error())
+		return
+	}
+
+	start := uint64(addr)
+	maxInstructionLength := s.debugger.Target().BinInfo().Arch.MaxInstructionLength()
+	offset := request.Arguments.InstructionOffset * maxInstructionLength
+	// Adjust the offset to include instructions before the requested address.
+	if offset < 0 {
+		start = uint64(addr + int64(offset))
```

Comment:
Unfortunately wn can not do this. For many (all?) ISA with variable instruction length the instruction encoding is not [self-synchronizing](https://en.wikipedia.org/wiki/Self-synchronizing_code), which means that this will usually produce a non-sensical disassembly.
There's also the problem that both start and end, as they are calculated, could cross a page bounary into unallocated memory causing the entire disassembly to fail even if there are enough instructions to satisfy the request.

The way the spec is pretty hard to satisfy here, the fully general interpretation would require backward disassembly, which is impossible to do. 
The easiest thing to do would be to call `Disassemble(-1, addr, 0)` which will disassemble the function around `addr` and then pad with undefined instructions in both directions. I don't know if this works with the requests sent by VSCode.